src/ch00-14-interfaces-traits.md

```diff
@@ -0,0 +1,37 @@
+# Contract interfaces and Traits generation
+
+Contract interfaces define the structure and behavior of a contract, serving as the contract's public ABI. They list all the function signatures that a contract exposes. For a detailed explanation of interfaces, you can refer to thee [Cairo Book](https://book.cairo-lang.org/ch99-01-02-a-simple-contract.html).
+
+In cairo, to specify the interface you need to define a trait annotated with `#[starknet::interface]` and then implement that trait in the contract.
+
+When a function needs to access the contract state, it must have a `self` parameter of type `ContractState`. This implies that the corresponding function signature in the interface trait must also take a `TContractState` type as a parameter. It's important to note that every function in the contract interface must have this `self` parameter of type `TContractState`.
+
+You can use the `#[generate_trait]` attribute to implicitly generate the trait for a specific implementation block. This attribute automatically generates a trait with the same functions as the ones in the implemented block, replacing the `self` parameter with a generic `TContractState` parameter.
+
+In summary, there's two ways to handle interfaces:
```

Comment:
> there's two ways to handle interfaces:

Technically, when traits generated by `#[generate_trait]` are not interfaces: they're just basic traits reflecting the content of the implementation and exposing the ContractState.